support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow/SingleLogoutPreparationAction.java

```diff
@@ -0,0 +1,79 @@
+package org.apereo.cas.support.pac4j.web.flow;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apereo.cas.util.Pac4jUtils;
+import org.apereo.cas.web.support.CookieRetrievingCookieGenerator;
+import org.apereo.cas.web.support.WebUtils;
+import org.pac4j.core.context.J2EContext;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileManager;
+import org.pac4j.core.store.Store;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+
+/**
+ * The purpose of this action is to prepare the PAC4J Profile Manager for Single Logout.
+ * 
+ * The Profile Manager keeps the profiles in request + session but the session has already been destroyed. This action should restore the
+ * profile from a long term storage - {@link Store} and populate the PAC4J Profile Manager with it.
+ * 
+ * This action should be called from the Logout web flow.
+ * 
+ * @author jkacer
+ * 
+ * @since 5.3.0
+ */
+public class SingleLogoutPreparationAction extends AbstractAction {
```

Comment:
Actions classes dont belong in a core module, I dont think.

Reply:
Hmm, the already existing actions are in the same module and package.
Have a look here: https://github.com/apereo/cas/tree/master/support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow
And project 'cas-server-support-pac4j-webflow' contains just a configurer and configuration.
I would rather leave all actions at the same place. Or do you plan to move the existing ones?